iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp

```diff
@@ -316,56 +319,85 @@ class AdjustDepthwiseFilterShape : public OpRewritePattern<mhlo::ConvOp> {
 // into
 //
 // %z = "BinaryElementwiseOpT"(%x, %y) : (%[[SHAPE_BEFORE_BCAST]], %[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_BEFORE_BCAST]]
-// %result = "mhlo.broadcast_in_dim"(%z) {} : (%[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_AFTER_BCAST]]
+// %result = "mhlo.broadcast_in_dim"(%z) {broadcast_dimensions = %[[BCAST_DIMS]]} : (%[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_AFTER_BCAST]]
 //
 // clang-format on
-template <typename BinaryElementwiseOpT>
-class ReorderBroadcastInDimOpAndBinaryElementwiseOp
-    : public OpRewritePattern<BinaryElementwiseOpT> {
+template <typename ElementwiseOpT>
+class ReorderBroadcastInDimOpAndElementwiseOp
+    : public OpRewritePattern<ElementwiseOpT> {
  public:
-  using OpRewritePattern<BinaryElementwiseOpT>::OpRewritePattern;
+  using OpRewritePattern<ElementwiseOpT>::OpRewritePattern;
 
-  LogicalResult matchAndRewrite(BinaryElementwiseOpT op,
+  LogicalResult matchAndRewrite(ElementwiseOpT op,
                                 PatternRewriter &rewriter) const override {
-    auto lhsType = op.lhs().getType().template dyn_cast<ShapedType>();
-    auto rhsType = op.rhs().getType().template dyn_cast<ShapedType>();
-    if (!lhsType || !rhsType || !lhsType.hasStaticShape() ||
-        lhsType != rhsType) {
-      return failure();
+    Operation *operation = op.getOperation();
+    // Sanity check: all elementwise ops shoule have 1 or more operand, only 1
+    // result, and no attribute.
+    assert(operation->getNumOperands() >= 1 &&
+           operation->getNumResults() == 1 && operation->getAttrs().empty());
+
+    // Verify if all operands are of same type and have static shape.
+    auto operands = operation->getOperands();
+    auto operandType = operands[0].getType().template dyn_cast<ShapedType>();
+    for (auto operand : operands.drop_front(1)) {
+      auto type = operand.getType().template dyn_cast<ShapedType>();
+      if (!type || !type.hasStaticShape() || type != operandType) {
+        return failure();
+      }
     }
```

Comment:
Dont need this check. The later check guarantees that all the opreands are broadcast_in_dim.

According to the op definition, the result type of  broadcast_in_dim is a static shape tensor.

https://github.com/tensorflow/mlir-hlo/blob/aedcea6acf37165294ae39659f0ddcc5cf1efa23/include/mlir-hlo/Dialect/mhlo/IR/hlo_ops.td#L821-L833